Client/Frontend/Extensions/ClientPickerViewController.swift

```diff
@@ -38,14 +38,28 @@ enum LoadingState {
 }
 
 class ClientPickerViewController: UITableViewController {
+    enum DeviceOrClient {
+        case client(RemoteClient)
+        case device(RemoteDevice)
+
+        var identifier: String? {
+            switch self {
+            case .client(let c):
+                return c.fxaDeviceId
+            case .device(let d):
+                return d.id 
+            }
+        }
+    }
+    var devicesAndClients = [DeviceOrClient]()
+
     var profile: Profile?
     var profileNeedsShutdown = true
 
     var clientPickerDelegate: ClientPickerViewControllerDelegate?
 
     var loadState = LoadingState.LoadingFromCache
-    var clients: [RemoteClient] = []
-    var selectedClients = NSMutableSet()
+    var selectedIdentifiers = Set<String>() // Stores DeviceOrClient.identifier
```

Comment:
nit: `selectedFxADeviceIDs` so it is clearer that this is the type of ID we're referring to

Reply:
The `identifier` is either a fxaDeviceId or a Client.id so I wasn't sure what to call this variable.

https://github.com/mozilla-mobile/firefox-ios/pull/4682/files#diff-8293d659f046df5bce6ea32062e140d7R45